pages/api/expenses.js

```diff
@@ -1,11 +1,20 @@
 import dbConnect from "../../lib/dbConnect";
 import Expense from "../../models/Expense";
+import { unstable_getServerSession } from "next-auth/next";
+import { authOptions } from "./auth/[...nextauth]";
 
 export default async function handler(request, response) {
+  const session = await unstable_getServerSession(
+    request,
+    response,
+    authOptions
+  );
+
   if (request.method === "POST") {
     try {
       await dbConnect();
-      const postData = JSON.parse(request.body);
+      const postDataNoUser = JSON.parse(request.body);
+      const postData = { ...postDataNoUser, userEmail: session.user.email };
```

Comment:
Just out of interest: Is there a reason why you don't write the email address already on the expenses.js inside the `body` of the request?

Reply:
As it's theoretically possible to manipulate data in the front end, I thought it might be more secure to cut that and manage user data in the back-end only. E.g. that a user cannot send data to the DB, using another email address, than he is logged in with.